File: HA8_4.ts

import * as readline from 'readline'

function CheckEven(iNo : number)
{
    let iDigit : number = 0
    let iMult : number = 1

    while(iNo !=0)
    {
        iDigit = iNo % 10;
        iMult = iMult * iDigit;
        iNo = Math.floor(iNo / 10);
    }
    return iMult;
}

async function main()
{
    const rl = readline.createInterface({
        input : process.stdin,
        output: process.stdout
    })

    let iValue : number = 0
    let iRet : number = 0

    iValue = parseInt(await askQuestion(rl, "Enter a number")) ||0

    iRet = CheckEven(iValue)
    console.log("Even Count is : "+ iRet)

    rl.close();
}
function askQuestion(rl: readline.Interface, question: string): Promise<string>
{
    return new Promise((resolve) =>{
        rl.question(question,(answer : any) =>{
            resolve(answer)
        })
    })
}
main()

File: HA8_4.cpp
//Write a program which accept number from user and return multiplication of all digits
//Input : 2395
//Output : 270

#include<iostream>
using namespace std;

class Digit
{
    public:
        int iNo;

    Digit(int X)
    {
        iNo = X;
    }

    int Multiple()
    {
        int iMult = 1;
        int iDigit = 0;

        while(iNo !=0)
        {
            iDigit = iNo % 10;
            iMult = iMult* iDigit;
            iNo = iNo / 10;
        }
        return iMult;
    }
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter a Number : "<<"\n";
    cin>>iValue;

    Digit dobj(iValue);

    iRet = dobj.Multiple();

    cout<<"Multiples of Digit are : "<<iRet<<"\n";

    return 0;    
}